src/ansys_sphinx_theme/extension/linkcode.py

```diff
@@ -0,0 +1,269 @@
+        try:
+            fn = inspect.getsourcefile(sys.modules[obj.__module__])
+        except Exception:
+            return None
+        return None
+
+    fn = op.relpath(fn, start=os.path.dirname(os.path.abspath(__file__)))
+    fn_components = op.normpath(fn).split(os.sep)
+
+    if not source_path:
+        module = modname.split(".")[0]
+        repo_index = fn_components.index(module)
+    else:
+        if source_path in fn_components:
+            repo_index = fn_components.index(source_path)
+        else:
+            module = modname.split(".")[0]
+            repo_index = fn_components.index(module)
+            fn_components.insert(repo_index, source_path)
+    fn = "/".join(fn_components[repo_index:])
+
+    try:
+        source, lineno = inspect.getsourcelines(obj)
+    except Exception:  # pragma: no cover
+        lineno = None
+
+    if lineno and not edit:
+        linespec = f"#L{lineno}-L{lineno + len(source) - 1}"
+    else:
+        linespec = ""
+
+    blob_or_edit = "edit" if edit else "blob"
+    return f"http://github.com/{library}/{blob_or_edit}/{github_version}/{fn}{linespec}"
+
+
+def link_code(app: Sphinx, doctree: Node):
+    """
+    Automatically add "View Source" links to the documentation.
+
+    This function is an event handler that automatically adds "View Source" links to the
+    documentation for Python, C, C++, and JavaScript objects. It is designed to work with
+    the Sphinx documentation tool.
+
+    Parameters
+    ----------
+    app : sphinx.application.Sphinx
+        Sphinx application instance.
+    doctree : docutils.nodes.Node
+        Document tree for the current page.
+
+    Notes
+    -----
+    The function uses the `sphinx_linkcode_resolve` function to resolve the link for each
+    documented object. The link is added as a "View Source" link to the documentation
+    when rendered.
+
+    For Python, the function extracts information such as module name and fullname from
+    the signature node to create the link.
+
+    The `link_code_library` configuration option can be used to specify a custom library
+    for link resolution.
+
+    Notes
+    -----
+    Adapted from the original `linkcode` extension in Sphinx, see
+    https://github.com/sphinx-doc/sphinx/blob/main/sphinx/ext/linkcode.py
+    """
+    env = app.builder.env
+    html_context = getattr(env.config, "html_context")
+    github_user = html_context.get("github_user", "")
+    github_repo = html_context.get("github_repo", "")
+    github_source = html_context.get("source_path", "")
+    github_version = html_context.get("github_version", "main")
+    if github_user and github_repo:
+        library = f"{github_user}/{github_repo}"
+    elif hasattr(env.config, "link_code_library") and hasattr(env.config, "link_code_source"):
+        library = getattr(env.config, "link_code_library")
+        github_source = getattr(env.config, "link_code_source")
+        github_version = getattr(env.config, "link_code_branch")
+
```

Comment:
The precedence behavior should be reflected here... Let's give precedence to the second if-branch. Also... you should also check that ``link_code_branch`` is defined. You are not doing it.

Reply:
Sure i will adapt that, link_code_branch is optional value, if is is not given, it will take main as default value